snuba/datasets/discover.py

```diff
@@ -0,0 +1,236 @@
+            ('username', Nullable(String())),
+            ('email', Nullable(String())),
+            ('ip_address', Nullable(String())),
+        ])
+
+        return common + self.get_events_only_columns() + self.get_transactions_only_columns()
+
+    def get_events_only_columns(self) -> ColumnSet:
+        return ColumnSet([
+            ('group_id', Nullable(UInt(64))),
+            ('primary_hash', Nullable(FixedString(32))),
+            ('type', Nullable(String())),
+            # Promoted tags
+            ('level', Nullable(String())),
+            ('logger', Nullable(String())),
+            ('server_name', Nullable(String())),
+            ('transaction', Nullable(String())),
+            ('site', Nullable(String())),
+            ('url', Nullable(String())),
+
+            ('message', Nullable(String())),
+            ('search_message', Nullable(String())),
+            ('title', Nullable(String())),
+            ('location', Nullable(String())),
+            ('culprit', Nullable(String())),
+            ('received', Nullable(DateTime())),
+            ('geo_country_code', Nullable(String())),
+            ('geo_region', Nullable(String())),
+            ('geo_city', Nullable(String())),
+            ('sdk_name', Nullable(String())),
+            ('sdk_integrations', Nullable(Array(String()))),
+            ('sdk_version', Nullable(String())),
+            ('version', Nullable(String())),
+            ('http_method', Nullable(String())),
+            ('http_referer', Nullable(String())),
+            ('exception_stacks.type', Nullable(String())),
+            ('exception_stacks.value', Nullable(String())),
+            ('exception_stacks.mechanism_type', Nullable(String())),
+            ('exception_stacks.mechanism_handled', Nullable(UInt(8))),
+            ('exception_frames.abs_path', Nullable(String())),
+            ('exception_frames.filename', Nullable(String())),
+            ('exception_frames.package', Nullable(String())),
+            ('exception_frames.module', Nullable(String())),
+            ('exception_frames.function', Nullable(String())),
+            ('exception_frames.in_app', Nullable(UInt(8))),
+            ('exception_frames.colno', Nullable(UInt(32))),
+            ('exception_frames.lineno', Nullable(UInt(32))),
+            ('exception_frames.stack_level', Nullable(UInt(16))),
+            ('modules.name', String()),
+            ('modules.version', String()),
+        ])
+
+    def get_transactions_only_columns(self) -> ColumnSet:
+        return ColumnSet([
+            ('trace_id', Nullable(UUID())),
+            ('span_id', Nullable(UInt(64))),
+            ('transaction_name', String()),
+            ('transaction_hash', Nullable(UInt(64))),
+            ('transaction_op', Nullable(String())),
+            ('start_ts', Nullable(DateTime())),
+            ('start_ms', Nullable(UInt(16))),
+            ('finish_ts', Nullable(DateTime())),
+            ('finish_ms', Nullable(UInt(16))),
```

Comment:
This is the same as timestamp on the common fields. It is up to us to decide whether to format that as datetime or as two fields

Reply:
Getting a single datetime value is more useful for sentry.

Reply:
Timestamp is being filled with the `finish_ts` value. I left `finish_ts` in as well though for transactions.

Reply:
What's the reason for keeping finish_ts ?

To give some context:
- the precision of clickhouse datetime columns is "seconds"
- that is why we added separate columns finish_ts/finish_ms 
- datetime in python do not have this constraint, we should not need two separate representations.

Reply:
@markstory Thinking over this, actually having a single datetime is not something we can do in the dataset definition, we need to work on the snuba reader for that. /snuba/reader.py. We should not try to achieve that in this PR, it is going to be complex.  This is because we do not have a dataset specific resultProcessor.
So I would keep timestamp + milliseconds as an int.

@lynnagara, I don't think we should leave the finish_ts field. (or maybe keep only the finish_ts). Duplicated columns are going to be confusing.

Reply:
Because of the way that time groupings work both fields are still referenced there. We have `time` and `bucketed_end` right now. It would be better to just have `time` but I don't have an easy fix for this.

Reply:
so the issue is that time group columns are managed in column_expr where we do not know much about dataset (technically we have access to the Query there, so we could do it properly). Anyway, why don't we define two grouping, one for finish_ts and one for timestamp, run a query processor first that would turn a grouped timestamp column into one of the two which will then be resovled by column_Expr ?

Reply:
The bucketed `time` grouping column doesn't need ms precision as we're always grouping on wider ranges.

We currently don't read the `finish_ms` directly out of snuba. The only time we use the millisecond precision is when we're rendering the span graph but all of that data is already coming out of nodestore. That is a roundabout way of saying I rescind my earlier comment and am ok with separate timestamp and ms columns.